src/Form/RenderFileTab.tsx

```diff
@@ -37,43 +43,66 @@ export default function RenderFileTab({ step, splitSchema, setSplitSchema }: Ren
 
   return (
     <Grid container justifyContent='center'>
-      <Stack direction='row' spacing={2} sx={{ p: 3 }}>
-        <Box sx={{ textAlign: 'center' }}>
-          <label htmlFor={codeId}>
-            <Typography sx={{ p: 1 }} variant='h5'>
-              Upload a code file (.zip)
-            </Typography>
-            <Input style={{ margin: '10px' }} id={codeId} type='file' onChange={handleCodeChange} accept='.zip' />
-            <Button variant='outlined' component='span'>
-              {code ? displayFilename(code.name) : 'Upload file'}
-            </Button>
-          </label>
-        </Box>
-        <Divider orientation='vertical' flexItem />
-        <Box sx={{ textAlign: 'center' }}>
-          <label htmlFor={binaryId}>
-            <Typography sx={{ p: 1 }} variant='h5'>
-              Upload a binary file (.zip)
-            </Typography>
-            <Input style={{ margin: '10px' }} id={binaryId} type='file' onChange={handleBinaryChange} accept='.zip' />
-            <Button variant='outlined' component='span'>
-              {binary ? displayFilename(binary.name) : 'Upload file'}
-            </Button>
-          </label>
-        </Box>
-      </Stack>
+      {buildOptionsStep !== undefined && buildOptionsStep.state.uploadType === ModelUploadType.Zip && (
+        <Stack direction='row' spacing={2} sx={{ p: 3 }}>
+          <Box sx={{ textAlign: 'center' }}>
+            <label htmlFor={codeId}>
+              <Typography sx={{ p: 1 }} variant='h5'>
+                Upload a code file (.zip)
+              </Typography>
+              <Input style={{ margin: '10px' }} id={codeId} type='file' onChange={handleCodeChange} accept='.zip' />
+              <Button variant='outlined' component='span'>
+                {code ? displayFilename(code.name) : 'Upload file'}
+              </Button>
+            </label>
+          </Box>
+          <Divider orientation='vertical' flexItem />
+          <Box sx={{ textAlign: 'center' }}>
+            <label htmlFor={binaryId}>
+              <Typography sx={{ p: 1 }} variant='h5'>
+                Upload a binary file (.zip)
+              </Typography>
+              <Input style={{ margin: '10px' }} id={binaryId} type='file' onChange={handleBinaryChange} accept='.zip' />
+              <Button variant='outlined' component='span'>
+                {binary ? displayFilename(binary.name) : 'Upload file'}
+              </Button>
+            </label>
+          </Box>
+        </Stack>
+      )}
+      {buildOptionsStep !== undefined && buildOptionsStep.state.uploadType === ModelUploadType.ModelCard && (
+        <Typography sx={{ p: 2 }}>Uploading a model card without any code or binary files</Typography>
+      )}
     </Grid>
   )
 }
 
 export function FileTabComplete(step: Step) {
-  return step.state.binary && step.state.code
+  const buildOptionsStep = step.state.steps.find(
```

Comment:
Should be able to wrap this with `useMemo` so that we only search through the array when it's changed